src/pages/Login/Login.js

```diff
@@ -1,7 +1,88 @@
+import {Button, FormControl, Link, Paper, Stack, TextField} from "@mui/material";
+import Box from "@mui/material/Box";
+import Typography from "@mui/material/Typography";
+import {useEffect, useState} from "react";
+import {useDispatch, useSelector} from "react-redux";
+import {useNavigate} from "react-router-dom";
+import {login} from "store/auth/authActions";
 import classes from "./Login.module.scss";
 
+const initialState = {
+    "username": {
+        "value": "",
+        "error":false
+    },
+    "password": {
+        "value": "",
+        "error":false
+    },
+}
+
 const Login = () => {
-    return <h1>Login page!</h1>
+    const [formState, setFormState] = useState(initialState);
+    const [loginSubmit, setLoginSubmit] = useState(false);
+    const [errorMessage, setErrorMessage] = useState('');
+    const {user, token, error} = useSelector(state => state.authReducer);
+
+    const dispatch = useDispatch();
+    const navigate = useNavigate();
+
+    useEffect(() => {
+        if(loginSubmit){
+            let error = "";
+            if (formState['username'].error || formState['password'].error){
+                error = "Please enter username and password!";
+            }
+            if (error === "") {
+                dispatch(login(formState['username'].value, formState['password'].value));
+            }
+            setErrorMessage(error);
+            setLoginSubmit(false);
+        }
+    }, [formState, loginSubmit]);
+
+    useEffect(() => {
+        if(user !== null) {
+            localStorage.setItem("user", JSON.stringify(user));
+            localStorage.setItem("token", token);
+            navigate('/');
+        }
+        else if (error !== '') {
+            setErrorMessage('Login failed');
+        }
+    }, [user, error]);
+
+    const updateField = (name, value) => {
+        const error = value.trim() === "";
+        setFormState( prevState => ({
+            ...prevState,
+            [name] : {"value": value, "error": error}
+        }))
+    }
+
+    const inputHandler = event => {
+        const {name , value} = event.target;
+        updateField(name, value);
+    }
+
+    const loginHandler = () => {
+        updateField("username", formState["username"].value);
+        updateField("password", formState["password"].value);
+        setLoginSubmit(true);
+    }
+
+    return <Box sx={{height:'500px', display:'flex', alignContent:'center', alignItems:'center', textAlign:'center', justifyContent:'center', padding:'20px'}}>
```

Comment:
you could make this content a bit wider as it seems too narrow at the moment. Maybe 350px wide